#VRML_SIM R2020a utf8
WorldInfo {
  basicTimeStep 16
}
Viewpoint {
  orientation 0.23689805020238563 -0.8107267055716678 -0.535351774708199 2.6521896841875403
  position -4.531845572977196 4.452699244611753 -1.7845687478667736
  followType "Mounted Shot"
}
TexturedBackground {
  texture "dusk"
}
TexturedBackgroundLight {
  texture "twilight_cloudy_empty"
}
Floor {
  rotation 0 0.7071067811865475 0.7071067811865475 0
  name "floor(1)"
  size 7 4
  appearance ChequeredParquetry {
    textureTransform TextureTransform {
    }
  }
}
AdvertisingBoard {
  translation 3.06 1.44 -3.12e-06
  rotation 0 1 0 1.5708
  backTexture [
    "textures/webots_billboard2.jpg"
  ]
}
DEF WALL_1 Solid {
  translation 0.5 0.05 0
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 0.5 0.05 0.97
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(5)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -1.43 0.05 0.97
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(7)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -2.39 0.05 1.56
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(23)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -2.39 0.05 0.65
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(24)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -2.39 0.05 -0.2
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(25)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -2.39 0.05 -1.11
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(26)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -2.39 0.05 -1.99
  rotation 0 1 0 1.5708
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(27)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -0.59 0.05 -1.99
  rotation 0 1 0 1.5708
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(29)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 0.29 0.05 -1.99
  rotation 0 1 0 1.5708
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(31)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -1.46 0.05 -1.99
  rotation 0 1 0 1.5708
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(30)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -2.39 0.05 -1.6
  rotation 0 1 0 1.5708
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(28)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -1.43 0.05 0.05
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(8)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation -1.43 0.05 -0.88
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(9)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 1.38 0.05 -1.5
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(15)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 2.42 0.05 -1.5
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(32)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 2.41 0.05 -0.56
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(33)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 2.41 0.05 0.34
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(34)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 2.41 0.05 1.3
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(35)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 1.9 0.05 1.75
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(36)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 1.9 0.05 -1.97
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(37)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 1.38 0.05 0.12
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(16)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 1.38 0.05 1.05
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(17)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 1.38 0.05 1.56
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(18)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_1 Solid {
  translation 0.5 0.05 -1.52
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  name "solid(14)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_2 Solid {
  translation -0.5 0.05 0
  children [
    USE WALL_SHAPE
  ]
  name "solid(1)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0 0.05 -0.5
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(2)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation -1.4 0.05 -0.5
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(12)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation -2.75481e-06 0.05 -1.25
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(10)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0.93 0.05 -1.98
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(13)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation -0.940003 0.05 -1.25
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(11)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation -0.59 0.05 0.5
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(3)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0.019998 0.05 1.47
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(4)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0.91 0.05 1.98
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(19)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation -0.06 0.05 1.98
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(20)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation -1 0.05 1.98
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(21)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation -1.95 0.05 1.98
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(22)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation -0.94 0.05 1.47
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(6)"
  boundingObject USE WALL_SHAPE
}
Robot {
  translation -1.135923139580149 0.03986048964925353 -0.11820110510985471
  rotation 1.4546022815695478e-08 -0.9999999999999998 1.1304878198362039e-08 -1.8201078840034512
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 4406.3533458531365
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation -0.6226573080185829 0.6226595951707838 -0.4739123392704681 4.026710868316934
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 0.038763298385178356 1.4028606263871623e-09 -0.009869473973094353
        angularVelocity -0.2467368365980192 -1.3094964182911952e-08 -0.9690825213044528
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 2613.2351050692128
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation -0.27065469287445226 0.27065568704647697 0.9238460565994941 1.6499271151535961
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 0.038763301330388376 -1.4029596517960506e-09 -0.00986947362723496
        angularVelocity -0.24673683674876715 -1.2989637210378678e-08 -0.9690825214939265
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 4406.404894896888
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation -0.629927013966903 0.6299293278222746 -0.45429175539967537 3.994424318170194
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 0.03876330036856055 1.4025959913361422e-09 -0.009869473324634972
        angularVelocity -0.24673683661024326 -1.3091293571470617e-08 -0.9690825214463399
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position 2613.42643223728
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation -0.18495118569423952 0.18495186505912647 0.9651869593614731 1.606226041532183
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 0.03876330335064415 -1.4032231840356847e-09 -0.009869472944065339
        angularVelocity -0.24673683677137395 -1.2995596262435935e-08 -0.9690825216354082
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.87079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.27079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "4_wheels_collision_avoidance"
  linearVelocity 0.03876330085869254 -1.8160517807087893e-13 -0.0098694734672572
  angularVelocity 2.3112752414398998e-08 1.626039068810068e-08 -5.881926408648352e-09
}
